hack/prepare.sh

```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -e
+
 git submodule init
 git submodule update
```

Comment:
Can you add `|| true` for the rest of the script after the git submodule init? Otherwise `set -e` will not work for my workflow anymore.

Reply:
To all of them? That seems not right, at that point, I would `set +e`.

Which commands are known/expected to fail?

Reply:
I need git submodule init and update separately. Can we refactor instead? prepare-repo and have a separate target for that? Then, we can keep `set -e`

Reply:
This patch is about using `set -e` and not silently ignoring errors, it does not question which commands to call.

Usually, `git submodule init` and `git submodule update` commands do not fail, if the directory is already initialized/updated. Could you explain the case where they are known to fail?

Reply:
From your comments I don't understand what the problem is (or what you suggest as solution).

This patch is about scripts not ignoring errors. Nothing more.

If you want to move code around or remove it entirely, then please either send a patch or elaborate on the problem.

Reply:
Can you rebase on top of https://github.com/openshift/dpu-operator/pull/198 ?

Reply:
I will do (once #198 is merged). Setting PR back to Draft for now (due to that).